invenio_records_rest/query.py

```diff
@@ -25,8 +25,9 @@ def default_search_factory(self, search, query_parser=None):
     """
     def _default_parser(qstr=None):
         """Default parser that uses the Q() from elasticsearch_dsl."""
+        fields = getattr(search, "boosted_fields", []) + ["*"]
```

Comment:
Given the issue with the `fields`/`lenient`, what about if we rewrite this? (and move it inside the if)
```
if qstr:
    boosted = getattr(search, "boosted_fields", [])
    extra_params = {}
    if boosted:
        extra_params["fields"] = boosted + ["*"]
        extra_params["lenient"] = True  # when listing fields that are not string (number, date) ignore formatting errors
    return Q("query_string", query=qstr, **extra_params) 
```

We have also to document this, otherwise the `boosted_fields` attr is very hidden.